### PR TITLE
Dan/java list spec fix

### PR DESCRIPTION
**Pull request by danmason**
Temporary fix allowing Java list types to be used within the Java API (in place of PersistentVector)

### CODE REVIEW

**Review by jarohen — CHANGES_REQUESTED** [review]
cheers Dan :)

@hraberg and I have been discussing our plans for how we'd like the existing Clojure/Java API to change in the near future, conclusions as follows:

* We currently have a mixed Clojure/Java API, where the Clojure implementation implements the API as defined by the Java interface
* You're obviously adding a Java adapter on top of this
* We'd like to reduce the current Clojure/Java low-level API to just a low-level Clojure API, suitable for Clojure-based tools to integrate with, and then your Java adapter calls into this. 
* Your Java adapter will be making this conversion from Java types to Clojure types
* With this in mind, I'd like the lifetime of this fix to be the same as the lifetime of the Java part of the low-level API - i.e. we don't want this fix to linger beyond the time we get rid of the Clojure/Java dance. 

tl;dr: could this fix be made in the Java part of the low-level API instead?

**Comment by danmason** [comment]
I've moved the conversion out of `api.clj` and into the java files, though not entirely certain on the placement - would it be worth adding a new class for this rather than placing it within `Crux.java`?

**Review by jarohen — APPROVED** [review]
on reflection, this was the right place for this functionality - I'd forgotten that there is no external Java implementation of the ICrux*API interfaces, so this is the outermost location to put this shim.

nice one @danmason :)
